lib/elixir/test/elixir/kernel_test.exs

```diff
@@ -233,20 +233,30 @@ defmodule KernelTest do
   end
 
   test "in/2 optimized" do
-    assert expand_to_string(quote(do: foo in [])) ==
-           "Enum.member?([], foo)"
-
-    assert expand_to_string(quote(do: rand() in 1..2)) =~
-           "var = rand()"
-    assert expand_to_string(quote(do: rand() in 1..2)) =~
-           ~S[:erlang.andalso(:erlang.is_integer(var), :erlang.andalso(:erlang.>=(var, 1), :erlang.=<(var, 2)))]
-
-    assert expand_to_string(quote(do: rand() in [1, 2])) =~
-           "var = rand()"
-    assert expand_to_string(quote(do: rand() in [1, 2])) =~
-           ~S[:erlang.or(:erlang.=:=(var, 2), :erlang.=:=(var, 1))]
-    assert expand_to_string(quote(do: rand() in [1 | [2]])) =~
-           ~S[:erlang.or(:erlang.=:=(var, 1), :erlang.=:=(var, 2))]
+    assert expand_to_string(quote(do: foo in [])) == "Enum.member?([], foo)"
+
+    expected =
+      "(fn var -> " <>
+      ":erlang.andalso(:erlang.is_integer(var), :erlang.andalso(:erlang.>=(var, 1), :erlang.=<(var, 2))) " <>
+      "end).(rand())"
+    assert expand_to_string(quote(do: rand() in 1..2)) == expected
+
+    expected =
+      "(fn var -> " <>
+      ":erlang.or(:erlang.=:=(var, 2), :erlang.=:=(var, 1)) " <>
+      "end).(rand())"
+    assert expand_to_string(quote(do: rand() in [1, 2])) == expected
+
+    expected =
+      "(fn var -> " <>
+      ":erlang.or(:erlang.=:=(var, 1), :erlang.=:=(var, 2)) " <>
+      "end).(rand())"
+    assert expand_to_string(quote(do: rand() in [1 | [2]])) == expected
+  end
+
+  response = %{code: 200}
+  if is_map(response) and response.code in 200..299 do
+    :pass
```

Comment:
@whatyouhide it needs to be outside of the function context, we can move it into a test but still with a module definition wrapping.